contracts/src/rollup/FallbackOperator.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.30;
+
+import { LineaRollupBase } from "./LineaRollupBase.sol";
+import { IFallbackOperator } from "./interfaces/IFallbackOperator.sol";
+/**
+ * @title Contract to manage EIP-4844 blob submission.
+ * @author ConsenSys Software Inc.
+ * @custom:security-contact security-report@linea.build
+ */
+abstract contract FallbackOperator is LineaRollupBase, IFallbackOperator {
+  /// @dev Keep 50 free storage slots for inheriting contracts.
+  uint256[50] private __gap_FallbackOperator;
+
+  /**
+   * @notice Initializes _FallbackOperator - used for new networks only.
+   * @param _fallbackOperator The initial computed genesis shnarf.
+   */
+  function __FallbackOperator_init(address _fallbackOperator) internal virtual {
+    require(_fallbackOperator != address(0), ZeroAddressNotAllowed());
+    fallbackOperator = _fallbackOperator;
+    emit FallbackOperatorAddressSet(msg.sender, _fallbackOperator);
+  }
+
+  /**
+   * @notice Sets the fallback operator role to the specified address if six months have passed since the last finalization.
+   * @dev Reverts if six months have not passed since the last finalization.
+   * @param _messageNumber Last finalized L1 message number as part of the feedback loop.
+   * @param _rollingHash Last finalized L1 rolling hash as part of the feedback loop.
+   * @param _lastFinalizedTimestamp Last finalized L2 block timestamp.
+   */
+  function setFallbackOperator(uint256 _messageNumber, bytes32 _rollingHash, uint256 _lastFinalizedTimestamp) external {
```

Comment:
(Outside PR Scope)

The function name `setFallbackOperator` suggests it sets a variable, but it actually grants the `OPERATOR_ROLE` to an already-stored address.

I'm wondering if `activateFallbackOperator` better reflects its semantics